framework_unittests/test_Server_2.py

```diff
@@ -0,0 +1,103 @@
+import unittest
+from server.server import Server
+from router.router import Router
+from multiprocessing import Process
+from server.ipc import IPC
+import time
+import os
+import socket
+
+
+def block_until_server_is_online():
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    print("wait", flush=True)
+    while not not sock.connect_ex(('localhost', 5000)):
+        time.sleep(3)
+        print('.', end="", flush=True)
+    sock.close()
+    print("server is online", flush=True)
+
+
+class ServerCore(object):
+
+    @classmethod
+    def tearDownClass(cls):
+        time.sleep(2)
+        print("Shutdown server")
+        ipc_client = IPC()
+        ipc_client.connect()
+        server_proxy = ipc_client.get_server_proxy()
+
+        # wait until server has all tasks before executed
+        for i in range(2):  # do it two times to be sure
+            routers = server_proxy.get_routers()
+            for router in routers:
+                while not not server_proxy.get_routers_task_queue_size(router.id):
+                    time.sleep(2)
+                    print('.', end="", flush=True)
+
+        server_proxy.stop()
+
+    def setUp(self):
+        self.ipc_client = IPC()
+        self.ipc_client.connect()
+        self.server_proxy = self.ipc_client.get_server_proxy()
+
+        # wait until server has all tasks before executed
+        for i in range(2):  # do it two times to be sure
+            routers = self.server_proxy.get_routers()
+            for router in routers:
+                while not not self.server_proxy.get_routers_task_queue_size(router.id):
+                    time.sleep(2)
+                    print('.', end="", flush=True)
+
+    def test_get_routers(self):
+        routers = self.server_proxy.get_routers()
+        assert len(routers) != 0
+        assert isinstance(routers[0], Router)
+
+    def test_little_self_check(self):
+        started = self.server_proxy.start_test(0, "ConnectionTest")
+
+        assert started
```

Comment:
`unittest.TestCase` als Basisklasse und dann
```python
self.assertTrue(self.server_proxy.start_test(0, "ConnectionTest"))
```

`tearDownClass` sollte mit der anderen Basisklasse auch ausgefuehrt werden. Bin mir aber nicht sicher wie ihr die tests (in diesem Ordner) gerade "discovered".